plugins/modules/ndo_l3out_bgp_peer.py

```diff
@@ -0,0 +1,735 @@
+
+        if import_route_map is not None:
+            mso_values["importRouteMapRef"] = mso_template.get_route_map(
+                "import_route_map",
+                tenant_id,
+                tenant_name,
+                import_route_map,
+                route_map_objects,
+            ).get("uuid", None)
+
+        if export_route_map is not None:
+            mso_values["exportRouteMapRef"] = mso_template.get_route_map(
+                "export_route_map",
+                tenant_id,
+                tenant_name,
+                export_route_map,
+                route_map_objects,
+            ).get("uuid", None)
+
+        # BGP Controls
+        bgp_controls_key_mapping = {
+            "allow_self_as": "allowSelfAS",
+            "override_as": "asOverride",
+            "disabled_peer_as_check": "disablePeerASCheck",
+            "next_hop_self": "nextHopSelf",
+            "send_community": "sendCommunity",
+            "send_extended_community": "sendExtendedCommunity",
+            "send_domain_path": "sendDomainPath",
+        }
+        bgp_controls_map = map_keys_to_new_dict(bgp_controls, bgp_controls_key_mapping)
+
+        # Peer Controls
+        peer_controls_key_mapping = {
+            "bfd": "bfd",
+            "disable_peer_connected_check": "disableConnectedCheck",
+        }
+        peer_controls_map = map_keys_to_new_dict(peer_controls, peer_controls_key_mapping)
+
+        # Address Type Controls
+        address_families_key_mapping = {
+            "multicast": "afMast",
+            "unicast": "afUcast",
+        }
+        address_families_map = map_keys_to_new_dict(address_families, address_families_key_mapping)
+
+        # Private AS Controls
+        private_as_controls_key_mapping = {
+            "remove_all": "removeAll",
+            "replace_with_local_as": "replaceWithLocalAS",
+        }
+        private_as_controls_map = map_keys_to_new_dict(private_as_controls, private_as_controls_key_mapping)
+        if private_as_controls_map and (private_as_controls_map.get("removeAll") or private_as_controls_map.get("replaceWithLocalAS")):
+            private_as_controls_map["removeExclusive"] = True
+
+        if not mso.existing:
+            mso_values["bgpControls"] = bgp_controls_map if bgp_controls_map is not None else None
+            mso_values["peerControls"] = peer_controls_map if peer_controls_map is not None else None
+            mso_values["addressTypeControls"] = address_families_map if address_families_map is not None else None
+            mso_values["privateASControls"] = private_as_controls_map if private_as_controls_map is not None else None
+            cleaned_mso_values = remove_none_values(mso_values)
+            mso.sanitize(cleaned_mso_values)
+            ops.append(dict(op="add", path=bgp_peer_path, value=cleaned_mso_values))
+
+        elif mso.existing:
+            proposed_payload = copy.deepcopy(mso.existing)
+            mso_values_remove = list()
```

Comment:
what is the use of this values remove list?

Reply:
`mso_values_remove` is used to store the attributes which need to be cleared from the existing object.